katello/hooks/pre_commit/09-version_locking.rb

```diff
@@ -0,0 +1,28 @@
+# evaluate version locking settings
+cli_param = app_value(:lock_package_versions)
+custom_config_value = get_custom_config(:lock_package_versions)
+lock_versions = cli_param.nil? ? !!custom_config_value : cli_param
+
+if lock_versions != custom_config_value
+  store_custom_config(:lock_package_versions, lock_versions)
+  kafo.config.configure_application
+end
+
+if lock_versions
+  unless system('command -v foreman-maintain > /dev/null')
+    Kafo::Helpers.log_and_say :error, 'Locking of package versions was requested but foreman-maintain is not installed'
+    kafo.class.exit 1
+  end
+  unless system('foreman-maintain packages -h > /dev/null 2>&1')
+    Kafo::Helpers.log_and_say :error, 'Locking of package versions was requested but foreman-maintain version installed does not support it'
+    kafo.class.exit 1
+  end
+end
+
+# unlock packages if locked
+if system('foreman-maintain packages is-locked --assumeyes > /dev/null')
```

Comment:
This still makes it a hard requirement. Should this be inside `if lock_versions`?

Reply:
@ekohl this is outside `lock_versions` intentionally for the case the packages were locked manually and auto-locking is not enabled in installer. What behavior of installer would you expect in such case?

It is not actually hard req as the condition is negative also when the command does not exist. We may add `2>&1` to hide the error out though.

Reply:
You're right, I totally missed that. Nice :)

Reply:
@ekohl I assume it is okay to let installer unlock manually locked packages, please let me know if it is not. I also suppressed the error out for clean output. Updated.

Reply:
IMHO as long as it's documented, that's fine. Users can still opt of out locking and handle it manually if I read the code correct. That's what we would advise users.